k8s-bench/README.md

```diff
@@ -8,10 +8,54 @@
 ```sh
 # build the k8s-bench binary
 go build
+```
+
+#### Run Subcommand
+
+The `run` subcommand executes the benchmark evaluations.
+
+```sh
+# Basic usage with mandatory output directory
+./k8s-bench run --agent-bin <path/to/kubectl-ai/binary> --output-dir .build/k8sbench
 
 # Run evaluation for scale related tasks
-./k8s-bench run --agent-bin <path/to/kubectl-ai/binary> --task-pattern scale --kubeconfig <path/to/kubeconfig>
+./k8s-bench run --agent-bin <path/to/kubectl-ai/binary> --task-pattern scale --kubeconfig <path/to/kubeconfig> --output-dir .build/k8sbench
+
+# Run evaluation for a specific LLM provider and model with tool use shim disabled
+./k8s-bench run --llm-provider=grok --models=grok-3-beta --agent-bin ../kubectl-ai --task-pattern=fix-probes --enable-tool-use-shim=false --output-dir .build/k8sbench
+
+# Run evaluation with all available options
+./k8s-bench run \
+  --agent-bin <path/to/kubectl-ai/binary> \
+  --kubeconfig ~/.kube/config \
+  --tasks-dir ./tasks \
+  --task-pattern fix \
+  --llm-provider gemini \
+  --models gemini-2.5-pro-preview-03-25,gemini-1.5-pro-latest \
+  --enable-tool-use-shim true \
+  --quiet true \
+  --output-dir .build/k8sbench
+```
 
+#### Available flags for `run` subcommand:
+
+| Flag | Description | Default | Required |
+|------|-------------|---------|----------|
+| `--agent-bin` | Path to kubectl-ai binary | - | Yes |
+| `--output-dir` | Directory to write results to | - | Yes |
+| `--tasks-dir` | Directory containing evaluation tasks | ./tasks | No |
+| `--kubeconfig` | Path to kubeconfig file | ~/.kube/config | No |
+| `--task-pattern` | Pattern to filter tasks (e.g. 'pod' or 'redis') | - | No |
+| `--llm-provider` | Specific LLM provider to evaluate (e.g. 'gemini' or 'ollama') | gemini | No |
+| `--models` | Comma-separated list of models to evaluate | gemini-2.5-pro-preview-03-25 | No |
+| `--enable-tool-use-shim` | Enable tool use shim | true | No |
+| `--quiet` | Quiet mode (non-interactive mode) | true | No |
+
+#### Analyze Subcommand
+
+The `analyze` subcommand processes results from previous runs:
+
+```sh
```

Comment:
Thanks for improving the docs. I did a very poor job with this README.md.

We have a helper script under the `ci` jobs in the top level directory to make it easy to run and analyze evals. It builds the current kubectl-ai and k8s-bench. Eventually, we plan to run these peridocially.

```shell
KUBECONFIG=~/.kube/auto-ai-1 TEST_ARGS="--enable-tool-use-shim=false --task-pattern=configure-" ./dev/ci/periodics/run-evals

./dev/ci/periodics/analyze-evals

```

We can probably add those instructions as well (may be in a separate PR).

/cc @tuannvm